UNIXFS.md

```diff
@@ -59,6 +64,27 @@ For files that are comprised of more than a single block, the 'Type' field will
 
 This data is serialized and placed inside the 'Data' field of the outer merkledag protobuf, which also contains the actual links to the child nodes of this object.
 
+## Metadata
+
+There are three ways to specify file metadata:
+
+1. The repeated `metadata` field in a directory applies metadata to each file in the directory.
+2. An intermediary node with a `Type` of `Metadata` applies metadata to an individual file. While this feature should be supported, it has been deprecated.
+3. An optional `defaultMetadata` field to specify the _default_ metadata for files in the directory. If unspecified, the default mode is 0755 and the default modification time is the epoch.
+  * Files: The default metadata is applied as-is.
```

Comment:
What is the default `mimeType` string? `""`? No default (field is optional anyway)?

Reply:
Yeah, please consider storing the entire UNIX 32bit st_mode structure. Or alternbatively just go with https://golang.org/pkg/os/#FileMode

Reply:
I agree. Including the actual file type is definitely a good idea.

Reply:
For those following along - this is what including the type ( i.e. full st_mode/mode_t ) allows. This is the same CID as the one used earlier in https://github.com/ipfs/specs/pull/223#discussion_r338504206. Observe how a user now has clear visual cues what is a directory and thus explorable. All can be displayed, [backwards-compatibly](https://ipfs.io/ipfs/bafybeif7ceybwlonoct3bnjtnnfookm2iezslo2ypwrsae6ilkc2ucvsfm/linux-5.3.7/), without downloading any extra blocks.

**Note: temporary hacked together gateway, will not persist**: http://5.9.79.235:2020/ipfs/bafybeif7ceybwlonoct3bnjtnnfookm2iezslo2ypwrsae6ilkc2ucvsfm/linux-5.3.7/